sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala

```diff
@@ -65,6 +66,20 @@ private[sql] trait SQLTestUtils extends SparkFunSuite with SQLTestUtilsBase with
     }
   }
 
+  /**
+   * A helper function for turning off/on codegen.
+   */
+  protected def testWithWholeStageCodegenOnAndOff(testName: String)(f: String => Unit): Unit = {
```

Comment:
not a big deal at all but just wanted to say. Looks we could just `f: () => Unit`. `codegenEnabled` as an argument is used at two places.

One place is just for clue and it can be removed since test name already describes if whole stage codegen is off or not.

The other place looks it needs but I think we can get read it from sql conf.